packages/doenetml-worker-rust/lib-doenetml-core/src/core/dependency.rs

```diff
@@ -348,6 +344,199 @@ pub fn create_dependencies_from_instruction_initialize_essential(
+                            || match_profiles.contains(&ComponentProfile::Text)
+                        {
+                            let essential_origin = EssentialDataOrigin::AttributeChild(
+                                attribute_name,
+                                essential_data_index,
+                            );
+
+                            // Essential data is from parent_idx, so it will be shared with the extend_source
+                            // if the children came from an extend_source
+                            let essential_data_view = if let Some(current_view) =
+                                essential_data[parent_idx].get(&essential_origin)
+                            {
+                                current_view.create_new_read_only_view()
+                            } else {
+                                let value = StateVarValue::String(string_value.clone());
+                                let new_view = create_essential_data_for(
+                                    parent_idx,
+                                    essential_origin.clone(),
+                                    InitialEssentialData::Single {
+                                        value,
+                                        used_default: false,
+                                    },
+                                    essential_data,
+                                );
+                                new_view.create_new_read_only_view()
+                            };
+
+                            dependencies.push(Dependency {
+                                source: DependencySource::Essential {
+                                    component_idx: parent_idx,
+                                    origin: essential_origin,
+                                },
+                                value: essential_data_view,
+                            });
+
+                            essential_data_index += 1;
+                        }
+                    }
+                    _ => (),
+                }
+            }
+
+            if dependencies.is_empty() {
+                // Found no matching attribute children.
+                // This means that the component and any component extend sources do not have any attribute children.
+
+                // Create an essential dependency with the default_value for the state variable
+                // Treat the essential data as though it came from the first string attribute child
+                // of the component, except recursing to extend source components
+                // in order to share the essential data with the extend source.
+
+                let source_idx =
+                    get_recursive_extend_source_component_when_exists(components, component_idx);
+
+                let essential_origin = EssentialDataOrigin::AttributeChild(attribute_name, 0);
+
+                let essential_data_view =
```

Comment:
I really wish we could rename `essential` to `primitive`

Reply:
I'm still not convinced by `primitive`. Maybe that's because I've used `primitive` in DoenetML more in reference to primitive data types when creating attribute primitives, where these attributes could only be primitive data types without dynamic features. The essential data does currently correspond to primitive data types, but it won't stay that way as we add more complicated features such as arrays.

If the term `essential` is going to be confusing, that's a strike against it. It'll just take more work to dislodge it from the essential place it has in my head when thinking about the DoenetML data structure.

Reply:
What about `RootVar`? As long as it doesn't get confused with the DAST root...or `DependencylessVar` or `NoDepData`?

I'm only one sample point, but to me `essential` means important and indispensable. But all the vars that are referenced are indispensable when computing the final state. The `essential` ones here just indicates where a calculation must start.